homeassistant/components/shelly/event.py

```diff
@@ -95,42 +102,69 @@ async def async_setup_entry(
         if TYPE_CHECKING:
             assert coordinator
 
-        key_instances = get_rpc_key_instances(coordinator.device.status, RPC_EVENT.key)
+        if config_entry.data[CONF_SLEEP_PERIOD] and not coordinator.device.initialized:
```

Comment:
I needed to add this to be able to add/restore entities when the device is not yet initialized (because sleepy devices).